pkg/clusteragent/admission/mutate/config/config.go

```diff
@@ -217,6 +232,53 @@ func injectionMode(pod *corev1.Pod, globalMode string) string {
 	return globalMode
 }
 
+// injectExternalDataEnvVar injects the External Data environment variable.
+// The format is: it-<init>,cn-<container_name>,pu-<pod_uid>
+func injectExternalDataEnvVar(pod *corev1.Pod) bool {
+	injected := false
+
+	type containerInjection struct {
+		container *corev1.Container
+		init      bool
+	}
+	var containerInjections []containerInjection
+
+	// Collect all containers and init containers
+	for i := range pod.Spec.Containers {
+		containerInjections = append(containerInjections, containerInjection{&pod.Spec.Containers[i], false})
+	}
+	for i := range pod.Spec.InitContainers {
+		containerInjections = append(containerInjections, containerInjection{&pod.Spec.InitContainers[i], true})
+	}
```

Comment:
not sure if we can avoid this overhead, especially that we are in a webhook. 

The alternative I am thinking about is having a helper function like this:

```
func injectExternalDataEnvVarToContainersList(ctrs []*corev1.Container, init bool) {
// Loop directly on ctrs and inject the env var
}
```

After that you can simplify the function you wrote into something like:
```
injectExternalDataEnvVarToContainersList(pod.Spec.Containers, false)
injectExternalDataEnvVarToContainersList(pod.Spec.InitContainers, true)
```

I guess this makes the code easier to read and debug, and also is slightly better in terms of performance (no need to define the new struct type and copy containers into slices)

WDYT?

Reply:
Checked together, current implementation ended up being the best option.